rfcs/rfc-005-branches-tags-and-channels.md

```diff
@@ -0,0 +1,52 @@
+```
+========================= master branch (edge channel) =======================>
+         \                      \                     \
+          \___v0.7.0 tag         \                     \
+           \                      \         v0.9.0 tag__\
+            \          v0.8.0 tag__\                     \
+ v0.7.1 tag__\                      \                 v0.9 branch (beta channel)
+              \___v0.7.2 tag         \___v0.8.1 tag
+               \                      \
+                \                      \
+           v0.7 branch         v0.8 branch (stable channel)
+
+```
+
+## Branches and Tags
+
+### master branch
+All new development occurs on the `master` branch.
+
+Bug fixes that affect a `vX.Y` branch are first made on `master`.  Once the bug
+fix lands on `master` it is cherry-picked into the `vX.Y` branch and potentially
+the `vX.Y+1` branch.  The exception to this rule is when a bug fix for `vX.Y`
+doesn't apply to `master` or `vX.Y+1`.
+
+Immediately after a new stabilization branch is forged, the minor version of the
+`master` branch is incremented.  Incrementing the major version of the `master`
+branch is outside the scope of this document.
+
+### v*X.Y* stabilization branches
+These are stabilization branches for a given milestone.  They are created off
+the `master` branch as late as possible prior to the milestone release.
+
+It is expected that only the latest and latest - 1 stabilization branches will
+be modified.  Older branches remain as a historical curiosity for future code
+archaeologists.
+
+### v*X.Y.Z* release tag
+The release tags are created as desired by the owner of the given stabilization
+branch, and cause that *X.Y.Z* release to be shipped to https://crates.io,
+https://snapcraft.io/, https://www.npmjs.com/ and elsewhere.
+
+Immediately after a new v*X.Y.Z* branch tag has been created the patch version
+number of the stabilization branch is incremented.
+
+## Channels
+Channels are used by end-users (humans and bots) to consume the branches
```

Comment:
The purpose to track the stability of releases. When end-users want to upgrade, they can choose to upgrade to the latest stable release, the latest experimental release (beta), or the unreleased bleeding edge. Miners will typically track `stable`, our performance-oriented dashboard will track `beta`, and developers will typically track `edge`.